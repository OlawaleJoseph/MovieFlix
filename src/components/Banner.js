import React from 'react';
import PropTypes from 'prop-types';
import {
  makeStyles, Grid, Typography, Button,
} from '@material-ui/core';

const Banner = ({ bgImg, desc }) => {
  const styles = makeStyles(theme => ({
    bg: {
      background: `url('${bgImg}') no-repeat center`,
      backgroundSize: 'cover',
      width: '100%',
      height: '450px',
      paddingTop: theme.spacing(6),
    },
    content: {
      height: '100%',
      paddingLeft: '30px',
    },
    btnContainer: {
      marginTop: '20px',
    },
    btn: {
      marginRight: '10px',
    },
  }));

  const classes = styles();

  return (
    <div className={classes.bg}>
      <Grid container alignItems="center" className={classes.content}>
        <Grid item xs={10} md={5} lg={4}>
          <Typography variant="body2" color="white">
            {desc}
          </Typography>
          <div className={classes.btnContainer}>
            <Button
              className={classes.btn}
              variant="outlined"
              size="large"
              color="secondary"
            >
              Play
            </Button>
            <Button
              className={classes.btn}
              variant="outlined"
              size="large"
              color="secondary"
            >
              Details
            </Button>
          </div>
        </Grid>
      </Grid>
    </div>
  );
};

Banner.propTypes = {
  bgImg: PropTypes.string.isRequired,
  desc: PropTypes.string.isRequired,
};

export default Banner;
