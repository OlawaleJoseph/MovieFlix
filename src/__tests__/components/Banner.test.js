import { Typography, Button } from '@material-ui/core';
import Banner from '../../components/Banner';
import shallowWrapper from '../../testSetup';

describe('Hero Section', () => {
  const movieInfo = {
    bgImg: 'imageUrl',
    desc: 'movie description',
  };

  const hero = shallowWrapper(Banner, movieInfo, true);

  test('should render Banner', () => {
    expect(hero).toBeTruthy();
    expect(hero).toMatchSnapshot();
  });

  test('should have a movie title and description', () => {
    const movieDescription = hero.find(Typography);

    expect(movieDescription).toHaveLength(2);
  });

  test('should render play button', () => {
    const buttons = hero.find(Button);

    expect(buttons).toHaveLength(2);
    expect(buttons.first().text()).toContain('Play');
  });

  test('should render play button', () => {
    const buttons = hero.find(Button);

    expect(buttons).toHaveLength(2);
    expect(buttons.last().text()).toContain('Details');
  });
});
