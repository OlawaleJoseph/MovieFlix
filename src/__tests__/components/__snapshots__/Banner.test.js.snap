// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Section should render Banner 1`] = `ShallowWrapper {}`;

exports[`Hero Section should render Navbar 1`] = `ShallowWrapper {}`;

exports[`Hero Section should render play button 1`] = `ShallowWrapper {}`;

exports[`Navigation bar should render Navbar 1`] = `ShallowWrapper {}`;
